deviceapi.py

```diff
@@ -0,0 +1,60 @@
+import json
+import requests
+from urllib.parse import urljoin
+from requests_toolbelt.multipart.encoder import MultipartEncoder
+
+class DeviceAPI:
+
+    def __init__(self, baseurl, devicename='uploader', password='password'):
+        self._baseurl = baseurl
+        self._devicename = devicename
+        self._token = self._get_jwt(password)
+        self._auth_header = {'Authorization': self._token}
+
+    def _get_jwt(self, password):
+        url = urljoin(self._baseurl, "/authenticate_device")
+        r = requests.post(url, data={
+            'devicename': self._devicename,
+            'password': password,
+            })
+
+        try: 
+            r.raise_for_status()
+        except:
+            raise Exception("Could not log on as '" + self._devicename +  "'.  Please check device name and password")
```

Comment:
As above.

There should probably be a base class which implements the JWT bits, or a standalone function that both API and DeviceAPI use to authenticate.